src/fdc3/dotnet/DesktopAgent/src/MorganStanley.ComposeUI.DesktopAgent/Fdc3DesktopAgent.cs

```diff
@@ -901,28 +837,54 @@ public async ValueTask<RaiseIntentResult<RaiseIntentResponse>> RaiseIntentForCon
             raiseIntentSpecification.TargetAppMetadata = appIntent.Apps.ElementAt(0);
             return await RaiseIntentToApplication(raiseIntentSpecification);
         }
-        
-        var resolverUIResult = await WaitForResolverUIAsync(appIntent.Apps);
+
+        var resolverUIResult = await _resolverUI.SendResolverUIRequest(appIntent.Apps);
         if (resolverUIResult != null && resolverUIResult.Error == null)
         {
             raiseIntentSpecification.TargetAppMetadata = (AppMetadata) resolverUIResult.AppMetadata!;
             return await RaiseIntentToApplication(raiseIntentSpecification);
         }
-        
+
         if (resolverUIResult?.Error != null)
         {
             return new()
             {
                 Response = RaiseIntentResponse.Failure(resolverUIResult.Error)
             };
         }
-        
+
         return new()
         {
             Response = RaiseIntentResponse.Failure(ResolveError.UserCancelledResolution)
         };
     }
 
+    private IEnumerable<AppIntent> FilterAppIntentsByAppId(IEnumerable<AppIntent> source, AppIdentifier appId)
```

Comment:
nitpick parameter names => AppIdentifier appIdentifier

Reply:
I don't see why should it be longer :)